themes/bootstrap/layouts/_default/tools.html

```diff
@@ -53,50 +53,32 @@
-            <legend>Languages</legend>
-            <div class="checkbox">
-              <label>
-                <input type="checkbox" value="all" data-bind="checked: selectedAllLanguages"/>
-                <em>All</em>
-              </label>
-            </div>
-            <div class="checkbox" data-bind="foreach: languages">
-              <label>
-                <input type="checkbox" data-bind="checkedValue: $data, checked: $parent.LanguagesSelected"/>
-                <span data-bind="text: $data"></span>
-              </label>
-              <br>
+            <legend>
+              <label id="languages-combo-label" class="combo-label">Languages</label>
+            </legend>
+            <div class="combo js-multiselect" id="js-multiselect-languages">
+              <ul class="selected-options" id="languages-combo-selected"></ul>
+              <input type="text" name="language-input" id="languages-combo" placeholder="" role="combobox" class="combo-input" autocomplete="off"/>
+              <div class="combo-menu" role="listbox" id="languages-listbox"></div>
             </div>
           </fieldset>
         </div>
         <div class="well">
           <fieldset>
-            <legend>Owners</legend>
-            <div class="checkbox">
-              <label>
-                <input type="checkbox" value="all" data-bind="checked: selectedAllOwners"/>
-                <em>All</em>
-              </label>
+            <legend>
+              <label id="owners-combo-label" class="combo-label">Owners</label>
+            </legend>
+            <div class="combo js-multiselect" id="js-multiselect-owners">
+              <ul class="selected-options" id="owners-combo-selected"></ul>
+              <input type="text" name="language-input" id="owners-combo" placeholder="" role="combobox" class="combo-input" autocomplete="off"/>
+              <div class="combo-menu" role="listbox" id="owners-listbox"></div>
             </div>
-            <div class="checkbox" data-bind="foreach: owners">
-              <label>
-                <input type="checkbox" data-bind="checkedValue: $data, checked: $parent.OwnersSelected"/>
-                <span data-bind="text: $data"></span>
-              </label>
-              <br>
-            </div >
           </fieldset>
         </div>
       </div>
       <div class="col-md-8 main">
-        <p class="lead">
-          There
-          <span data-bind="visible: filteredTools().length != 1">are</span>
-          <span data-bind="visible: filteredTools().length == 1">is</span>
-          <span id="total_tools" data-bind="text: filteredTools().length">0</span>
-          <span data-bind="visible: filteredTools().length != 1">tools/extensions</span>
-          <span data-bind="visible: filteredTools().length == 1">tool/extension</span>
-          matching your search:
+        <p class="lead" id="search-result-message">
+          <!-- There are N tools/extensions matching your search: -->
```

Comment:
Out of curiosity: `There <span id="search-result-message"> matching your search:` is not possible?

Reply:
Decided to initialize this with a `loading` message, and then update the whole message on state change. Reason: with the proposed `<span>`, the message's length changes in a jarring way on search.